activation/handler_test.go

```diff
@@ -1065,6 +1073,78 @@ func TestHandler_ProcessAtx_OwnNotMalicious(t *testing.T) {
 	require.Nil(t, proof)
 }
 
+func TestHandler_PublishesPostMalfeasanceProofs(t *testing.T) {
+	goldenATXID := types.ATXID{2, 3, 4}
+	atxHdlr := newTestHandler(t, goldenATXID)
+
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	nodeID := sig.NodeID()
+
+	proof, err := identities.GetMalfeasanceProof(atxHdlr.cdb, nodeID)
+	require.ErrorIs(t, err, sql.ErrNotFound)
+	require.Nil(t, proof)
+
+	ch := newChallenge(0, types.EmptyATXID, goldenATXID, 1, &goldenATXID)
+	ch.InitialPost = &types.Post{}
+	nipost := newNIPostWithChallenge(t, types.HexToHash32("0x3333"), []byte{0x76, 0x45})
+
+	atx := newAtx(t, sig, ch, nipost.NIPost, 100, types.GenerateAddress([]byte("aaaa")))
+	atx.NodeID = &nodeID
+	vrfNonce := types.VRFPostIndex(0)
+	atx.VRFNonce = &vrfNonce
+	atx.SetEffectiveNumUnits(100)
+	atx.SetReceived(time.Now())
+	require.NoError(t, SignAndFinalizeAtx(sig, atx))
+	_, err = atx.Verify(0, 100)
+	require.NoError(t, err)
+
+	var got types.MalfeasanceGossip
+	atxHdlr.mclock.EXPECT().CurrentLayer().Return(atx.PublishEpoch.FirstLayer())
+	atxHdlr.mValidator.EXPECT().VRFNonce(gomock.Any(), goldenATXID, gomock.Any(), gomock.Any(), gomock.Any())
+	atxHdlr.mValidator.EXPECT().
+		Post(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any())
+	atxHdlr.mockFetch.EXPECT().RegisterPeerHashes(gomock.Any(), gomock.Any())
+	atxHdlr.mockFetch.EXPECT().GetPoetProof(gomock.Any(), atx.GetPoetProofRef())
+	atxHdlr.mValidator.EXPECT().InitialNIPostChallenge(&atx.NIPostChallenge, gomock.Any(), goldenATXID)
+	atxHdlr.mValidator.EXPECT().PositioningAtx(goldenATXID, gomock.Any(), goldenATXID, atx.PublishEpoch)
+	atxHdlr.mValidator.EXPECT().
+		NIPost(gomock.Any(), gomock.Any(), goldenATXID, atx.NIPost, gomock.Any(), atx.NumUnits, gomock.Any()).
+		Return(0, &verifying.ErrInvalidIndex{Index: 2})
+	atxHdlr.mtortoise.EXPECT().OnMalfeasance(gomock.Any())
+	atxHdlr.mpub.EXPECT().Publish(gomock.Any(), pubsub.MalfeasanceProof, gomock.Any()).DoAndReturn(
+		func(_ context.Context, _ string, data []byte) error {
+			require.NoError(t, codec.Decode(data, &got))
+			postVerifier := NewMockPostVerifier(gomock.NewController(t))
+			postVerifier.EXPECT().Verify(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("invalid"))
```

Comment:
What is the reason to have this in the callback?

Reply:
Validating POST malfeasance proof on line `nodeID, err := malfeasance.Validate(` requires checking if the POST proof in the malicious ATX is invalid (it checks only the one label that the proof reports).